panic: runtime error: invalid memory address or nil pointer dereference

When running `netlifyctl init manual`
it asks for the username (gitlab for this project)
it then pauses with no feedback

when pressing enter I get this error
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x165cf2b]

goroutine 1 [running]:
github.com/netlify/netlifyctl/vendor/github.com/xanzy/go-gitlab.(*Client).NewRequest(0xc420192000, 0x189f111, 0x4, 0x18a2b00, 0x7, 0x16ff200, 0xc4203daba0, 0xc420421680, 0x1, 0x1, ...)
        /home/david/src/github.com/netlify/netlifyctl/vendor/github.com/xanzy/go-gitlab/gitlab.go:298 +0x4b
github.com/netlify/netlifyctl/commands/init.getSession(0xc4203daba0, 0xc420421680, 0x1, 0x1, 0x0, 0x0, 0x1, 0xc4204e4570)
        /home/david/src/github.com/netlify/netlifyctl/commands/init/gitlab.go:131 +0xbc
github.com/netlify/netlifyctl/commands/init.newGitLabConfigurator(0x21f10d0, 0xc4204e4210, 0xc42051e100, 0xc4204e43f0, 0xc42051e100, 0xc420241038)
        /home/david/src/github.com/netlify/netlifyctl/commands/init/gitlab.go:33 +0x180
github.com/netlify/netlifyctl/commands/init.loadConfigurator(0x21f10d0, 0xc4204e4210, 0xc42051e100, 0x1010199, 0xc4204e4210, 0x30, 0x30)
        /home/david/src/github.com/netlify/netlifyctl/commands/init/git.go:39 +0x113
github.com/netlify/netlifyctl/commands/init.initSite(0x21f10d0, 0xc4204e4210, 0xc4204926c0, 0xc4201b9280, 0x1, 0x1, 0x1, 0xc4204e4180)
        /home/david/src/github.com/netlify/netlifyctl/commands/init/init.go:38 +0x6e6
github.com/netlify/netlifyctl/commands/middleware.ClientMiddleware.func1(0x21f10d0, 0xc4204e40f0, 0xc4204926c0, 0xc4201b9280, 0x1, 0x1, 0x142370f, 0x17c1b20)
        /home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:183 +0x339
github.com/netlify/netlifyctl/commands/middleware.AuthMiddleware.func1(0x21f10d0, 0xc4204e4090, 0xc4204926c0, 0xc4201b9280, 0x1, 0x1, 0x1d9e9e0, 0x19214b8)
        /home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:128 +0x1ca
github.com/netlify/netlifyctl/commands/middleware.LoggingMiddleware.func1(0x20097d0, 0xc420034088, 0xc4204926c0, 0xc4201b9280, 0x1, 0x1, 0xc4202a3560, 0xc420421b80)
        /home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:116 +0x1f1
github.com/netlify/netlifyctl/commands/middleware.DebugMiddleware.func1(0x20097d0, 0xc420034088, 0xc4204926c0, 0xc4201b9280, 0x1, 0x1, 0x0,0x0)
        /home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:89 +0x4bb
github.com/netlify/netlifyctl/commands/middleware.NewRunFunc.func1(0xc4204926c0, 0xc4201b9280, 0x1, 0x1, 0x0, 0x0)
        /home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:52 +0xf2
github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra.(*Command).execute(0xc4204926c0, 0xc4201b9240, 0x1, 0x1, 0xc4204926c0, 0xc4201b9240)
        /home/david/src/github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra/command.go:644 +0x3db
github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1d8d7c0, 0x1dbb7fd, 0x189eb65, 0x3)
        /home/david/src/github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra/command.go:734 +0x2d4
github.com/netlify/netlifyctl/commands.Execute()
        /home/david/src/github.com/netlify/netlifyctl/commands/netlify.go:41 +0x29b
main.main()
        /home/david/src/github.com/netlify/netlifyctl/main.go:16 +0x79
```

--- a/commands/init/gitlab.go
+++ b/commands/init/gitlab.go
@@ -2,10 +2,12 @@ package init
 
 import (
 	"bufio"
+	"errors"
 	"fmt"
 	"net/url"
 	"os"
 	"os/signal"
+	"strings"
 	"syscall"
 
 	"github.com/netlify/netlifyctl/context"
@@ -25,15 +27,21 @@ func newGitLabConfigurator(ctx context.Context, url *url.URL) (*gitlabConfigurat
 		return nil, err
 	}
 
-	password, err := getPassword()
+	password, err := getPassword(url.Host)
 	if err != nil {
 		return nil, err
 	}
 
-	session, _, err := getSession(&gitlab.GetSessionOptions{
-		Login:    gitlab.String(username),
-		Password: gitlab.String(password),
-	}, nil)
+	u := strings.TrimSpace(username)
+	p := strings.TrimSpace(password)
+	if u == "" || p == "" {
+		return nil, errors.New("Username and password cannot be blank")
+	}
+
+	session, _, err := getSession(url.String(), &gitlab.GetSessionOptions{
+		Login:    gitlab.String(u),
+		Password: gitlab.String(p),
+	})
 	if err != nil {
 		return nil, err
 	}
@@ -95,7 +103,9 @@ func getUsername(host string) (string, error) {
 }
 
 // copied from GitHub's hub
-func getPassword() (string, error) {
+func getPassword(host string) (string, error) {
+	fmt.Printf("%s password: ", host)
+
 	stdin := int(syscall.Stdin)
 	initialTermState, err := terminal.GetState(stdin)
 	if err != nil {
@@ -126,8 +136,9 @@ func getPassword() (string, error) {
 	return string(passBytes), nil
 }
 
-func getSession(opt *gitlab.GetSessionOptions, options ...gitlab.OptionFunc) (*gitlab.Session, *gitlab.Response, error) {
-	client := &gitlab.Client{}
+func getSession(urlStr string, opt *gitlab.GetSessionOptions, options ...gitlab.OptionFunc) (*gitlab.Session, *gitlab.Response, error) {
+	client := gitlab.NewClient(nil, "")
+	client.SetBaseURL(urlStr)
 	req, err := client.NewRequest("POST", "session", opt, options)
 	if err != nil {
 		return nil, nil, err

